src/legacy/core_plugins/kibana/public/dashboard/panel/dashboard_panel.tsx

```diff
@@ -17,127 +17,162 @@
+  EmbeddableState,
+} from 'ui/embeddable';
+import { EmbeddableActions, EmbeddableErrorAction } from '../actions';
+import { PanelState } from '../selectors';
 import { PanelError } from './panel_error';
+import { PanelHeader } from './panel_header/panel_header';
+
+interface DashboardPanelProps {
+  intl: any;
+  viewOnlyMode: boolean;
+  onPanelFocused: (panel: string) => {};
+  onPanelBlurred: (panel: string) => {};
+  error: string | object;
+  destroy: () => void;
+  containerState: ContainerState;
+  embeddableFactory: EmbeddableFactory;
+  embeddableStateChanged: (embeddableStateChanges: EmbeddableState) => {};
+  embeddableIsInitializing: EmbeddableActions;
+  embeddableIsInitialized: (embeddableIsInitialized: EmbeddableMetadata) => void;
+  embeddableError: EmbeddableErrorAction;
+  initialized: boolean;
+  panel: PanelState;
+  isPopoverOpen: boolean;
+  isLoading: boolean;
+  calloutTitle: string;
+  className: any;
+}
 
-import {
-  EuiPanel,
-} from '@elastic/eui';
+interface State {
+  error: string | null;
+}
 
-class DashboardPanelUi extends React.Component {
-  constructor(props) {
+export class DashboardPanelUi extends React.Component<DashboardPanelProps, State> {
+  [panel: string]: any;
+  public mounted: boolean;
+  public embeddable!: Embeddable;
+  constructor(props: DashboardPanelProps) {
     super(props);
     this.state = {
-      error: props.embeddableFactory ? null : props.intl.formatMessage({
-        id: 'kbn.dashboard.panel.noEmbeddableFactoryErrorMessage',
-        defaultMessage: 'No factory found for embeddable',
-      }),
+      error: props.embeddableFactory
+        ? null
+        : props.intl.formatMessage({
+            id: 'kbn.dashboard.panel.noEmbeddableFactoryErrorMessage',
+            defaultMessage: 'No factory found for embeddable',
+          }),
     };
-
     this.mounted = false;
   }
 
-  async componentDidMount() {
+  public async componentDidMount() {
     this.mounted = true;
     const {
       initialized,
       embeddableFactory,
-      embeddableIsInitializing,
       panel,
       embeddableStateChanged,
       embeddableIsInitialized,
-      embeddableError,
     } = this.props;
-
     if (!initialized) {
-      embeddableIsInitializing();
-      embeddableFactory.create(panel, embeddableStateChanged)
-        .then((embeddable) => {
+      embeddableFactory
+        .create(panel, embeddableStateChanged)
+        .then((embeddable: Embeddable) => {
           if (this.mounted) {
             this.embeddable = embeddable;
             embeddableIsInitialized(embeddable.metadata);
             this.embeddable.render(this.panelElement, this.props.containerState);
           } else {
-            embeddable.destroy();
+            this.embeddable.destroy();
           }
         })
-        .catch((error) => {
+        .catch(error => {
           if (this.mounted) {
-            embeddableError(error.message);
+            this.embeddableError = error.message;
```

Comment:
don't think you want to change this to a local variable, the function was setting in the redux state tree

Reply:
Ah ok thanks! Changed it back